contributed_definitions/nyaml/NXsample_component_set.yaml

```diff
@@ -0,0 +1,37 @@
+category: base
+doc: |
+  Set of sample components and their configuration. The idea here is to store all descriptors that 
+  cannot be stored on the level of the individual sample component.
+symbols:
+  doc: |
+    DOC MISSING.
+  n_comp: |
+    number of components
+type: group
+NXsample_component_set(NXobject):
+  layer(NX_CHAR):
+    doc: |
+      Number of layers of the sample (e.g. bulk, monolayer, pentalayer, etc.)
+  concentration(NX_FLOAT):
```

Comment:
concentration should really be 1/volume while composition is count / count the first is NX_ANY the second is NX_DIMENSIONLESS

When you would like to insists on NX_MASS_DENSITY that is a composition with mass weight factorization as I mentioned several times pint can at the moment not distinguish this from atomic % (at.-%) but we have to give the opportunity I agree to have all these possibilities and they are so frequently in use that we should add them here